### PR TITLE
CI: Add stale skip utility

### LINKED ISSUE #422: Add a utility to check for stale skips
Some of the tests are skipped due to issues that cause them to fail.

The problem is, those "skips" may no longer be valid as the issue may have been resolved.

Create a simple script to look for stale skips.

### PR DESCRIPTION
**Pull request by jodh-intel**
Created a simple script that looks for test skips that are no longer
valid as the issue they are referencing is closed.

Fixes #422.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
This is RFC as the technique used to check if the github issue is closed is horrid.

Can anyone think of a better way that doesn't involve rewriting this in go and having to deal with all the auth complexities of 2FA?

FTR, running this script shows that some of our skips are indeed stale:

```
INFO: Found 8 tests containing skips
INFO: Found 9 skip URLs
INFO: Checking skip URL https://github.com/clearcontainers/runtime/issues/1042
INFO: Remove skip from following files (issue https://github.com/clearcontainers/runtime/issues/1042 is closed):
INFO: 
INFO:     integration/docker-compose/docker-compose.bats
INFO: 
INFO: Checking skip URL https://github.com/clearcontainers/runtime/issues/868
INFO: Checking skip URL https://github.com/clearcontainers/runtime/issues/876
INFO: Checking skip URL https://github.com/kata-containers/agent/issues/138
INFO: Remove skip from following files (issue https://github.com/kata-containers/agent/issues/138 is closed):
INFO: 
INFO:     integration/cri-o/cri-o.sh
INFO: 
INFO: Checking skip URL https://github.com/kata-containers/agent/issues/190
INFO: Remove skip from following files (issue https://github.com/kata-containers/agent/issues/190 is closed):
INFO: 
INFO:     integration/docker/cap_test.go
INFO: 
INFO: Checking skip URL https://github.com/kata-containers/agent/issues/261
INFO: Checking skip URL https://github.com/kata-containers/runtime/issues/366
INFO: Checking skip URL https://github.com/kata-containers/shim/issues/46
INFO: Remove skip from following files (issue https://github.com/kata-containers/shim/issues/46 is closed):
INFO: 
INFO:     integration/docker/user_test.go
INFO: 
INFO: Checking skip URL https://github.com/kata-containers/tests/issues/264
```

/cc @chavafg, @GabyCT if you want to investigate.

**Comment by grahamwhaley** [comment]
Heh. I'd rather we used the GitHub REST API if we could. Are you intending to hook this into every CI build/run btw? If so, then yes, I could see the rate limit being an issue. If no, and we only expect to run this occasionally, then I suspect we would be fine with the rate limited API.
Another thought - if we are going to run this under the Jenkins CIs, then I think there is a way for Jenkins to hand the GitHub auth token into the CI environment, which maybe this script could then check for and use if available. /cc @chavafg

**Comment by jodh-intel** [comment]
I was waiting for that comment :)

I'll prolly leave this here as a POC for others to work on if they choose. Feel free to close if it's not useful.

**Comment by GabyCT** [comment]
@jodh-intel , I will check those skip issues and test them to see that we do not have failures

**Comment by jodh-intel** [comment]
Thanks @GabyCT! If we aren't comfortable with the approach this PR takes for checking the github page, I can close it as this was just a quick experiment which seemed to work :)

File: .ci/kata-find-stale-skips.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+# This is a crude heuristic and may break at any time. However, using the
+# github API is equally problematic - unauthenticated access is rate-limited
+# and authenticated access is difficult (2FA for example).
+is_github_issue_closed()
+{
+	local -r issue="$1"
+
+	# Look for the magic github token denoting the issue is still open
+	curl -sL "$issue"|grep -q 'div class="State State--green"' && echo "no" && return
```
Comment (chavafg) [inline .ci/kata-find-stale-skips.sh]: what about converting `$issue` to an API url, then curl and grep for `"state": "open"`?
```
issue=$(echo $issue | sed -e 's|github.com|api.github.com/repos|')

```

For the authentication, we can use a @katabuilder token so we do not have problems when retrieving the information

Comment (jodh-intel) [reply .ci/kata-find-stale-skips.sh]: I like it! Users still might get caught by the github rate-limit, but it's atleast now more reliable when it *does* run :)

Branch updated.

**Comment by jodh-intel** [comment]
Branch updated to resolve merge conflict.

**Comment by chavafg** [comment]
lgtm

**Comment by jodh-intel** [comment]
Hi @chavafg, @grahamwhaley - I'll leave it up to you, but we nominally require doc team approval for the 1 line doc change on this PR. Do you think this qualifies for the "fast track" process? :smile:

**Comment by chavafg** [comment]
I think it does ;)

**Comment by jodh-intel** [comment]
/me looks away and whistles to himself...

**Comment by grahamwhaley** [comment]
Sure. And my ack hits the docs team tickbox I believe... waiting for CIs...

**Pull request merged**